contracts/eos/MultiConverter/MultiConverter.cpp

```diff
@@ -348,26 +347,28 @@ void MultiConverter::mod_account_balance(name owner, symbol_code converter_curre
         });
 }
 
-void MultiConverter::modreserve(name owner, asset quantity, symbol_code converter_currency_code, name code, bool setup) {
+void MultiConverter::modreserve(name owner, asset quantity, symbol_code converter_currency_code, name code) {
+    require_auth(owner);
     reserves reserves_table(get_self(), converter_currency_code.raw());
     const auto& reserve = reserves_table.get(quantity.symbol.code().raw(), "reserve doesn't exist");
 
     converters converters_table(get_self(), converter_currency_code.raw());
     const auto& converter = converters_table.get(converter_currency_code.raw(), "converter does not exist");
-    check((!setup && converter.enabled) || !converter.launched, "converter was launched, or must be enabled");
 
     if (quantity.amount > 0)
         check(code == reserve.contract, "wrong origin contract for quantity");
-    else if (setup) // if quantity < 0 && setup == true
+    else
         action(
```

Comment:
we can get to that flow without checking any permissions over the contract itself (we just check that the owner is the caller) - doesn't it mean that the caller can withdraw the reserve?